### PR TITLE
Upgrade license to GPL-3.0-or-later

### PR DESCRIPTION
**Pull request by ssoloff**
Per #2764.

Things of note:

* I placed the license notice (with the exceptions) in the README.  This seemed simpler than creating a separate file (e.g. LICENSE-NOTICE) that no one may actually read.
* I added a copyright notice to the license notice, as recommended in the GPL.  I attributed the copyright to "TripleA contributors."  Not sure if this is the best choice, but it really becomes moot if we adopt a CLA or a DCO (see #2880).
* The only exception I listed in the license notice was JavaMail.  After further research, I concluded that it is probably unnecessary to list exceptions for libraries that we do not actually _distribute_ with our release artifacts (which includes the JUnit 5 and JUnit 5 Samples libraries).  If anyone disagrees, it's a simple matter to add the four additional libraries to the exceptions table.
* I believe the license badge in the README will automatically update once this PR is merged.

Unrelated to #2764:

* I changed the Markdown headings in the README to use `#` delimiters instead of underlining them with `===`.
* I updated many links in the README to be canonical (e.g. use `www.triplea-game.org` instead of `triplea-game.github.io`), to use HTTPS if possible, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,40 +3,58 @@
 
 TripleA is a free game engine that runs on open source and is community supported.
 
-Installing TripleA and Playing
-==============================
-- Download and install TripleA: http://triplea-game.github.io/download/
+# Installing TripleA and Playing
+
+- Download and install TripleA: http://www.triplea-game.org/download/
```
Comment (RoiEXLab) [inline README.md]: `www.triplea-game.org` should be just `triplea-game.org`

Comment (ssoloff) [reply README.md]: 442d8fa9f

File: README.md [hunk 2]
```diff
@@ -3,40 +3,58 @@
...
+
+- Please see: https://github.com/triplea-game/lobby#bots
+
+# Development
 
-Development
-===========
 - Project documentation, including 'how to get started guides' at: https://github.com/triplea-game/triplea/tree/master/docs/dev
 - Broken map list: https://github.com/triplea-game/triplea/wiki/Broken-Maps
 - Additional feature request list: https://github.com/triplea-game/triplea/issues?q=label%3A%22ice+box+-+revisit+later%22+is%3Aclosed (list is not to be added to, if picked up, re-open and remove the ice-box label)
```
Comment (RoiEXLab) [inline README.md]: Unrelated, but the Ice Box label was renamed to `ice box - close and revisit later`, so this link is broken.
Additionally this link should be hidden behind a nicer name, it really does not look good in the mark down.

Comment (ssoloff) [reply README.md]: Agreed.  In general, I don't like dropping bare URLs in Markdown, especially ones as busy as the ice box link.  How about if I just hide _all_ links in the README behind a label?  For example:

# Development

* [Project documentation](https://github.com/triplea-game/triplea/tree/master/docs/dev) (including 'how to get started guides')
* [Broken map list](https://github.com/triplea-game/triplea/wiki/Broken-Maps)
* [Additional feature request list](https://github.com/triplea-game/triplea/issues?q=label%3A%22ice+box+-+revisit+later%22+is%3Aclosed) (list is not to be added to, if picked up, re-open and remove the ice-box label)

Comment (RoiEXLab) [reply README.md]: 👍 For consistency

Comment (ssoloff) [reply README.md]: 0752369aa

Thanks for catching the broken ice box link.  I tested all the others; not sure why I skipped that one. 🤷‍

**Comment by ssoloff** [comment]
> I believe the license badge in the README will automatically update once this PR is merged.

Confirmed.

**Pull request merged**